### PR TITLE
Controler and routes

### PR DESCRIPTION
**Pull request by GraysonHoward**
I think this is probably most of the routes we need.

### CODE REVIEW

**Comment by qpco** [comment]
need route for login page. also separate controller classes for each entity

File: src/main/java/dev/group1/Potlukk/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    // Get a user
+    @GetMapping("/user/{id}")
+    public User getUserByID(@PathVariable int id){
+        return null;
+    }
+
+    // Update User
+    @PutMapping("/users{id}")
```
Comment (qpco) [inline src/main/java/dev/group1/Potlukk/controller/UserController.java:24]: missing forward slash here

File: src/main/java/dev/group1/Potlukk/controller/ItemController.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    @PatchMapping("/items/{id}/needed")
+    public Item patchNeeded(@PathVariable int id){
+        return null;
+    }
+
+    @PatchMapping("/items/{id}/fulfilled")
+    public Item patchFulfilled(@PathVariable int id){
+        return null;
+    }
```
Comment (qpco) [inline src/main/java/dev/group1/Potlukk/controller/ItemController.java:37]: maybe combine routes into one and replace status with another path variable?

Comment (GraysonHoward) [reply src/main/java/dev/group1/Potlukk/controller/ItemController.java]: Good idea

**Review by qpco — APPROVED** [review]

File: src/main/java/dev/group1/Potlukk/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    // Update User
+    @PutMapping("/users/{id}")
+    @ResponseBody
+    public User putUser(@PathVariable int id){
+        return null;
+    }
+
+    // Remove a user
+    @DeleteMapping("/delete/{id}")
```
Comment (ZanettaR) [inline src/main/java/dev/group1/Potlukk/controller/UserController.java:31]: Any reason for not using user/id or users/id for the delete like the other routes in this controller?

File: src/main/java/dev/group1/Potlukk/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+@Component
+@RestController
+public class UserController {
+    // Add new user
+    @PostMapping("/users")
+    @ResponseBody
+    public User postUser(){
+        return null;
+    }
```
Comment (ZanettaR) [inline src/main/java/dev/group1/Potlukk/controller/UserController.java:15]: Could we use all plural or all singular naming conventions? Like how the items controller is uniform.

File: src/main/java/dev/group1/Potlukk/controller/UserController.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+import dev.group1.Potlukk.entities.User;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.*;
+
+@Component
+@RestController
+public class UserController {
+    // Add new user
+    @PostMapping("/users")
```
Comment (ZanettaR) [inline src/main/java/dev/group1/Potlukk/controller/UserController.java:11]: Could we use all plural (users) or all singular (user) naming conventions? Like how the items controller is uniform.

Comment (ZanettaR) [reply src/main/java/dev/group1/Potlukk/controller/UserController.java:11]: Same for the PotlukkController

**Review by ZanettaR — APPROVED** [review]

**Comment by ZanettaR** [comment]
What do we think about a route to get all users?

**Comment by ZanettaR** [comment]
Even though they're just mapped and not implemented yet?

**Review by ZanettaR — APPROVED** [review]

File: src/main/java/dev/group1/Potlukk/controller/ItemController.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+public class ItemController {
+
+    // Add an item to a potlukk
+    @PostMapping("/items")
+    public Item postItem(){
+        return null;
+    }
+
+    // Update Item
+    @PutMapping("/items{id}")
```
Comment (ZanettaR) [inline src/main/java/dev/group1/Potlukk/controller/ItemController.java:19]: Missing /

Comment (ZanettaR) [reply src/main/java/dev/group1/Potlukk/controller/ItemController.java:19]: Other than that good to go @GraysonHoward

**Pull request merged**